src/PhpDoc/ResolvedPhpDocBlock.php

```diff
@@ -824,6 +837,21 @@ private static function mergeOneParentReturnTag(?ReturnTag $returnTag, self $par
 
 		$parentType = $parentReturnTag->getType();
 
+		if ($classReflection !== null) {
+			$parentType = TypeTraverser::map(
+				$parentType,
+				static function (Type $type, callable $traverse) use ($classReflection): Type {
+					if ($type instanceof StaticType) {
+						return $type->changeBaseClass($classReflection);
```

Comment:
Great, I'd fix this the same way so 👍 

I just hate adding new properties to this class but I didn't figure out a better solution. yet. Merging this.